File: leetcode/js/largest-local-values-in-a-matrix.js
/**
 * @param {number[][]} grid
 * @return {number[][]}
 */
var largestLocal = function (grid) {
  const findMaxNum3x3 = (startI, startJ) => {
    let result = 0;

    for (let i = startI; i < startI + 3; i++) {
      for (let j = startJ; j < startJ + 3; j++) {
        result = Math.max(result, grid[i][j]);
      }
    }

    return result;
  };

  const result = [];

  for (let i = 0; i < grid.length; i++) {
    const row = [];

    for (let j = 0; j < grid[0].length; j++) {
      if (i + 2 < grid.length && j + 2 < grid[0].length) {
        row.push(findMaxNum3x3(i, j));
      }
    }

    if (row.length > 0) {
      result.push(row);
    }
  }

  return result;
};

File: leetcode/go/largest-local-values-in-a-matrix.go
package main

func LargestLocal(grid [][]int) [][]int {
	n := len(grid)
	result := make([][]int, 0)

	for i := range n - 2 {
			row := make([]int, 0)

			for j := range n - 2 {
					maxVal := 0

					for x := range 3 {
							for y := range 3 {
									maxVal = max(maxVal, grid[i + x][j + y])
							}
					}

					row = append(row, maxVal)
			}

			result = append(result, row)
	}

	return result
}